### PR TITLE
address golint warnings in package control

### PR DESCRIPTION
**Pull request by candysmurf**

### CODE REVIEW

File: control/config.go [hunk 1]
```diff
@@ -55,6 +55,7 @@ type config struct {
 	Plugins *pluginConfig `json:"plugins"`
 }
 
+// NewConfig creates a new snap plugin configuration
```
Comment (danielscottt) [inline control/config.go]: NewConfig returns a reference to a global config type for the snap daemon.

File: control/control.go [hunk 2]
```diff
@@ -125,21 +130,27 @@ type managesSigning interface {
 	ValidateSignature([]string, string, []byte) error
 }
 
-type ControlOpt func(*pluginControl)
+// PluginControlOpt is the type of pluginControl struture
```
Comment (danielscottt) [inline control/control.go]: PluginControlOpt is used to set optional parameters on the pluginControl struct

File: control/control.go [hunk 3]
```diff
@@ -125,21 +130,27 @@ type managesSigning interface {
...
+type PluginControlOpt func(*pluginControl)
 
-func MaxRunningPlugins(m int) ControlOpt {
+// MaxRunningPlugins returns the plugin control with the
```
Comment (danielscottt) [inline control/control.go]: The PluginControlOpt for setting the maximum number of plugins to run per pool

File: control/control.go [hunk 4]
```diff
@@ -125,21 +130,27 @@ type managesSigning interface {
...
+// maximum number of plugins set.
+func MaxRunningPlugins(m int) PluginControlOpt {
 	return func(c *pluginControl) {
 		maximumRunningPlugins = m
 	}
 }
 
-func CacheExpiration(t time.Duration) ControlOpt {
+// CacheExpiration returns the plugin control with the
```
Comment (danielscottt) [inline control/control.go]: CacheExpiration is the PluginControlOpt which sets the global metric cache TTL

File: control/plugin_manager.go [hunk 5]
```diff
@@ -238,21 +246,24 @@ func newPluginManager(opts ...pluginManagerOpt) *pluginManager {
 
 type pluginManagerOpt func(*pluginManager)
 
+// OptSetPluginConfig returns pluginManagerOpt strut
```
Comment (danielscottt) [inline control/plugin_manager.go]: OptSetPluginConfig sets the config on the plugin manager

File: control/plugin_manager.go [hunk 6]
```diff
@@ -485,6 +496,7 @@ func (p *pluginManager) UnloadPlugin(pl core.Plugin) (*loadedPlugin, serror.Snap
 	return plugin, nil
 }
 
+// GenerateArgs returns a new generated argument
```
Comment (danielscottt) [inline control/plugin_manager.go]: GenerateArgs generates the cli args to send when stating a plugin

File: control/monitor.go [hunk 7]
```diff
@@ -22,10 +22,12 @@ package control
 import "time"
 
 const (
+	// MonitorStopped - enum representation of monitor stopped state
 	MonitorStopped monitorState = iota - 1 // default is stopped
+	// MonitorStarted - enum representation of monitor started state
 	MonitorStarted
 
-	// Changed to one second until we get proper control of duration runtime into this.
+	// DefaultMonitorDuration - the default monior duration.
```
Comment (geauxvirtual) [inline control/monitor.go]: monitor

File: control/config.go [hunk 8]
```diff
@@ -55,6 +55,7 @@ type config struct {
 	Plugins *pluginConfig `json:"plugins"`
 }
 
+// NewConfig returns a reference to a global config type for the snap daemon
```
Comment (tiffanyfj) [inline control/config.go]: Would want other people to comment, but maybe something about how it initializes plugin config. It does a pass a reference, but mores specifically it's creating a config struct for someone to add config data to. 
@danielscottt @geauxvirtual

File: control/monitor.go [hunk 9]
```diff
@@ -98,7 +100,7 @@ func (m *monitor) Start(availablePlugins *availablePlugins) {
 	m.State = MonitorStarted
 }
 
-// stop the monitor
+// Stop the monitor
```
Comment (tiffanyfj) [inline control/monitor.go]: Either 
Stop stops the monitor  (to keep with how you're doing everything else)
or 
Stops the monitor

File: control/mttrie.go [hunk 10]
```diff
@@ -70,7 +70,7 @@ func NewMTTrie() *MTTrie {
 	return &MTTrie{m}
 }
 
-// Handy print out of the tr(i)e
+// String prints out of the tr(i)e
```
Comment (tiffanyfj) [inline control/mttrie.go]: Git says to ask you, @lynxbat. Why the (i)? As in it could be tree or trie?

File: control/router.go [hunk 11]
```diff
@@ -17,14 +17,16 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-// Router is the entry point for execution commands and routing to plugins
+// Package control Router is the entry point for execution commands and routing to plugins
```
Comment (tiffanyfj) [inline control/router.go]: I'd remove the first Router.

File: control/mttrie.go (hunk 10 above)
Comment (lynxbat) [inline control/mttrie.go]: Ask @danielscottt

File: control/runner.go [hunk 12]
```diff
@@ -43,11 +43,14 @@ var (
 type availablePluginState int
 
 const (
+	// HandlerRegistrationName - it is the control runner
```
Comment (tiffanyfj) [inline control/runner.go]: Just to keep with formatting, maybe make this one like the next 3?

File: control/mttrie.go (hunk 10 above)
Comment (danielscottt) [inline control/mttrie.go]: re-___trie___-val: https://en.wikipedia.org/wiki/Trie

File: control/config.go (hunk 8 above)
Comment (candysmurf) [inline control/config.go]: incorporated the feedback

**Comment by danielscottt** [comment]
This one looks ready to me.

**Pull request merged**